contractsV5/kyberDAO/KyberDAO.sol

```diff
@@ -0,0 +1,649 @@
+
+        uint perInPrecision = getStakerRewardPercentageInPrecision(staker, epoch);
+        require(perInPrecision > 0, "claimReward: No reward to claim");
+
+        require(feeHandler.claimReward(staker, epoch, perInPrecision), "claimReward: feeHandle failed to claim reward");
+
+        hasClaimedReward[staker][epoch] = true;
+        emit RewardClaimed(staker, epoch, perInPrecision);
+    }
+
+    // get latest network fee data + expiry data
+    // also save winning option data and latest network fee result
+    function getLatestNetworkFeeDataWithCache() public returns(uint feeInBps, uint expiryBlockNumber) {
+        uint curEpoch = getCurrentEpochNumber();
+
+        feeInBps = latestNetworkFeeResult;
+        // expiryBlockNumber = START_BLOCK + curEpoch * EPOCH_PERIOD - 1;
+        expiryBlockNumber = START_BLOCK.add(curEpoch.mul(EPOCH_PERIOD)).sub(1);
+
+        // there is no camp for epoch 0
+        if (curEpoch == 0) {
+            return (feeInBps, expiryBlockNumber);
+        }
+
+        uint campID = networkFeeCamp[curEpoch.sub(1)];
+        if (campID == 0) {
+            // not have network fee campaign, return latest result
+            return (feeInBps, expiryBlockNumber);
+        }
+
+        uint winningOption;
+        (winningOption, feeInBps) = getCampaignWinningOptionAndValue(campID);
+        // save latest winning option data
+        winningOptionData[campID] = encodeWinningOptionData(winningOption, true);
+
+        if (winningOption == 0) {
+            // no winning option, fall back to previous result
+            feeInBps = latestNetworkFeeResult;
+        } else {
+            // update latest result based on new winning option
+            latestNetworkFeeResult = feeInBps;
+        }
+    }
+
+    // return latest burn/reward/rebate data, also affecting epoch + expiry block number
+    function getLatestBRRData()
+        public
+        returns(uint burnInBps, uint rewardInBps, uint rebateInBps, uint epoch, uint expiryBlockNumber)
+    {
+        epoch = getCurrentEpochNumber();
+        // expiryBlockNumber = START_BLOCK + curEpoch * EPOCH_PERIOD - 1;
+        expiryBlockNumber = START_BLOCK.add(epoch.mul(EPOCH_PERIOD)).sub(1);
+        uint brrData = latestBrrResult;
+        if (epoch > 0) {
+            uint campID = brrCampaign[epoch.sub(1)];
+            if (campID != 0) {
+                uint winningOption;
+                (winningOption, brrData) = getCampaignWinningOptionAndValue(campID);
+                // save latest winning option data
+                winningOptionData[campID] = encodeWinningOptionData(winningOption, true);
+                if (winningOption == 0) {
+                    // no winning option, fallback to previous result
+                    brrData = latestBrrResult;
+                } else {
+                    // concluded campaign, updated new latest brr result
+                    latestBrrResult = brrData;
+                }
+            }
+        }
+
```

Comment:
Yes, it gets the network fee campaign of previous epoch, and checked if the winning option has been concluded.
In side `getCampaignWinningOptionAndValue` it will first check if the winning option data can be used, otherwise it will use formula to get winning option id